autogpt_platform/backend/backend/data/graph.py

```diff
@@ -355,6 +355,26 @@ def from_db(graph: AgentGraph):
             },
         )
 
+    @staticmethod
+    def _process_node(node: AgentNode, hide_secrets: bool) -> Node:
+        node_dict = node.model_dump()
+        if hide_secrets and "constantInput" in node_dict:
+            constant_input = json.loads(node_dict["constantInput"])
+            Graph._hide_secrets_in_input(constant_input)
+            node_dict["constantInput"] = json.dumps(constant_input)
+        return Node.from_db(AgentNode(**node_dict))
+
+    @staticmethod
+    def _hide_secrets_in_input(input_data: Dict[str, Any]):
+        sensitive_keys = ["api_key", "password", "token", "secret"]
```

Comment:
```suggestion
        sensitive_keys = ["credentials", "api_key", "password", "token", "secret"]
```

Reply:
This looks to expose a weakness in our system here. We shouldn't need to be guessing which fields are secret based on there names, rather using the `SecretField` in the blocks schema to define them.

out of scope of this fix though